ros/src/waypoint_updater/waypnt_updater.py

```diff
@@ -473,49 +626,62 @@ def set_waypoints_velocity(self):
         if self.waypoints[self.final_waypoints_start_ptr].get_v() == 0.0:
             # we are stopped
             offset = 0
-            stopping_distance = 0.0
             if self.state != 'stopped':
-                rospy.loginfo("We have stopped.")
+                rospy.logwarn("We have stopped.")
                 self.state = 'stopped'
+                self.stopping_distance = 0.0
+                self.min_stop_distance = 0.01
         else:
-            stopping_distance = get_accel_distance(
-                self.waypoints[self.final_waypoints_start_ptr].get_v(),
-                0.0, self.default_accel/accel_ratio)
-
-        rospy.loginfo("dist_to_tl = {:4.3f}, stopping_dist = {:4.3f}, state = {}"
-                      .format(dist_to_tl, stopping_distance, self.state))
+            if (math.fabs(self.prev_step_v -
+                    self.waypoints[self.final_waypoints_start_ptr].get_v()) > 0.1
+                    or 
+                    math.fabs(self.prev_step_a -
+                    self.waypoints[self.final_waypoints_start_ptr].get_a()) > 0.1):
```

Comment:
Is this condition just to reduce unnecessary recalculations of the stopping distances until reaching the next waypoint where the speed or accel is changing, especially for the maintain speed case? If so, could you add a comment to explain this for reference?